src/cargo/ops/cargo_new.rs

```diff
@@ -182,3 +202,20 @@ fn global_config(config: &Config) -> CargoResult<CargoNewConfig> {
         version_control: vcs,
     })
 }
+
+#[cfg(test)]
+mod tests {
+    use super::strip_rust_affixes;
+
+    #[test]
+    fn affixes_stripped() {
+        assert_eq!(strip_rust_affixes("rust-foo"), "foo");
+        assert_eq!(strip_rust_affixes("foo-rs"), "foo");
+        assert_eq!(strip_rust_affixes("rs_foo"), "foo");
+        // Only one affix is stripped
```

Comment:
I wrote an alternate version that performs that check, but the code ended up too hard to understand.

In practice it shouldn't matter anyway. I scanned crates.io a few days ago and there are zero packages that have more than one `rust`/`rs` affix.